### PR TITLE
Automate generating `RevenueCat-Swift.h` for SPM

### PR DESCRIPTION
**Pull request by NachoSoto**
- Fixes [sc-11746]
- See https://github.com/RevenueCat/purchases-ios/blob/362f023bf06f11efdf5b45db9d8f133c47e499d6/docs/V4_API_Updates.md#objc--spm
- Bug report: https://bugs.swift.org/browse/SR-15154

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-  1. Input new version number
-  2. Update changelog
-  3. A new branch and PR will automatically be created
-3. Merge PR when approved
-4. Make a tag and push, the rest will be performed automatically by CircleCI. If the automation fails, you can revert to manually calling `bundle exec fastlane deploy`.
+    1. Input new version number
+    2. Update changelog
+    3. A new branch and PR will automatically be created
+3. Generate new `RevenueCat-Swift.h` [for SPM](https://github.com/RevenueCat/purchases-ios/blob/main/docs/V4_API_Updates.md#objc--spm):
+    1. Run `bundle exec fastlane compile_autogenerated_header`
```
Comment (taquitos) [inline RELEASING.md:8]: I would even be happy to have this run on each deploy without having a separate lane. That way there's no thinking/possibility of missing it. We'll get tiny metadata updates, but I think that's fine.

Comment (NachoSoto) [reply RELEASING.md:8]: Then we'd also have to `git add && git commit && git push`, unless we also automate that, which I agree would be nice.

Comment (taquitos) [reply RELEASING.md:8]: I'm totally cool with that. I think we do that with the deploy script already for the version bumping.

Comment (NachoSoto) [reply RELEASING.md]: I was't familiar with the v4 release process cause I used the 3.x a few weeks ago, but now I am!
This is so much simpler 👍  and `bump` already does the right thing.

File: RELEASING.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #### Releasing:
-1. Create a `fastlane/.env` file with GitHub API token (see `fastlane/SAMPLE.env`)
+1. Create a `fastlane/.env` file with GitHub API token (see `fastlane/.env.SAMPLE`)
```
Comment (NachoSoto) [inline RELEASING.md:2]: This was wrong 😇

File: fastlane/Fastfile [hunk 3]
```diff
@@ -238,6 +240,33 @@ platform :ios do
     )
   end
 
+  desc "Creates RevenueCat-Swift.h for a new release"
+  private_lane :compile_autogenerated_header do |options|
```
Comment (NachoSoto) [inline fastlane/Fastfile:244]: I made this `private` now.

**Review by taquitos — APPROVED** [review]
🎈🐐

**Pull request merged**